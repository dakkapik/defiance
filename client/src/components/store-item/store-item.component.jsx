import React from "react";
import Button from "@material-ui/core/Button";
import "./store-item.styles.scss";
import { connect } from "react-redux";

import { setConnectedStore } from "../../redux/stores/stores.action";
import { SocketInitStore } from "../../redux/socket/socket.action";
export const StoreItem = ({
  storeInfo,

  setConnectedStore,
  SocketInitStore,
}) => {
  return (
    <div className="storeitem">
      {/* <div className="storeitem storeitem--text"></div> */}
      <Button
        onClick={() => {
          setConnectedStore(storeInfo);
          SocketInitStore(storeInfo);
        }}
        variant="outlined"
        color="inherit"
      >
        {storeInfo.name}
      </Button>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  setConnectedStore: (info) => dispatch(setConnectedStore(info)),
  SocketInitStore: (info) => dispatch(SocketInitStore(info)),
});

export default connect(null, mapDispatchToProps)(StoreItem);
