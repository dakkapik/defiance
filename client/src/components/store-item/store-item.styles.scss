.storeitem {
  margin: 5px;
  text-align: center;

  &--text {
    @extend .storeitem;
    color: white;
    font-size: 16px;
    margin-bottom: 20px;
  }
}
