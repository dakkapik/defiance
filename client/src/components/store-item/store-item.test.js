import React from "react";
import { StoreItem } from "./store-item.component";
import { shallow } from "enzyme";

describe("StoreItem component render", () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;
  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();
    const mockProps = {
      storeInfo: {
        imageUrl: "www.testImage.com",
        price: 10,
        name: "hats",
        quantity: 2,
      },
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
      clearItem: mockClearItem,
    };
    wrapper = shallow(<StoreItem {...mockProps} />);
  });

  it("should render StoreItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
