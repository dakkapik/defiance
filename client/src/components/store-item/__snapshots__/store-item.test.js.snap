// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreItem component should render StoreItem component 1`] = `ShallowWrapper {}`;
